platform/darwin/src/MGLStyleLayer.h

```diff
@@ -29,7 +29,10 @@ NS_ASSUME_NONNULL_BEGIN
  
  After initializing and configuring the style layer, add it to a map view’s
  style using the `-[MGLStyle addLayer:]` or
- `-[MGLStyle insertLayer:belowLayer:]` method.
+ `-[MGLStyle insertLayer:belowLayer:]` method. Should not be used to create an 
+ instance of MGLStyleLayer directly. Instead, create instances of
+ `MGLBackgroundStyleLayer` and the concrete subclasses of
+ `MGLForegroundStyleLayer`.
```

Comment:
This change will get overwritten the next time someone runs `make style-code-darwin`. This documentation is defined in [style-spec-overrides-v8.json](https://github.com/mapbox/mapbox-gl-native/blob/fc9f0ad9427fd61397f4d6eede4a6e5b746d088c/platform/darwin/scripts/style-spec-overrides-v8.json).